programs/survival/fakeGui.sc

```diff
@@ -0,0 +1,430 @@
+// 原作者: Scarpet
+// 修改: 猴子
+// 建議: 企鵝
+
+__config() -> {
+    'scope' -> 'global',
+    'stay_loaded' -> true,
+    'requires' -> {
+        'carpet' -> '>=1.4.57'
+    },
+    'commands' -> {
+        '' -> _() -> print('\nHow to use:\n Right click on the fake player\nor:\nExecute command /fakegui <player>'),
+        '<player>' -> _(fakeplayer) -> (
+            fakeplayer = player(fakeplayer);
+            if (
+                fakeplayer~'player_type' != 'fake' && player() != fakeplayer,
+                return(display_title(player(), 'actionbar', '§4§Lwait he\'s not a fake player...')),
+                __page('menu', player(), fakeplayer),
+            );
+        ),
+    },
+    'arguments' -> {
+        'player' -> {'type' -> 'players', 'single' -> true}
+    },
+};
+
+global_slotmap = [[-1,7],[-2,0],[-3,1],[-4,2],[-5,3]];
+global_fakeplayers_selected_slot = m();
+global_fakeplayersscreen = m();
+global_counts = m();
+global_models = l(
+    nbt('{ name: \'once\', title: \'§f§Lone click\', item: \'minecraft:minecart\'}'),
+    nbt('{ name: \'continuous\', title: \'§f§Lkeep clicking\', item: \'minecraft:apple\'}'),
+    nbt('{ name: \'interval\', title: \'§f§Ltime click\', item: \'minecraft:repeater\'}'),
+);
+
+__page(type_, creativeplayer, fakeplayer) -> (
+    // 菜單
+    if (type_ == 'menu', (
+        screen = create_screen(creativeplayer, 'generic_9x3', fakeplayer~'command_name'+' the menu', _(screen, player, action, data, outer(fakeplayer)) -> (
+            if(action == 'close', (
+                global_fakeplayersscreen:fakeplayer = null;
+                drop_item(screen, -1);
+                return();
+            ));
+            if (action == 'pickup', (
+                slot = data:'slot';
```

Comment:
You can combine multiple if statements into one, to avoid the return() calls, cos they are slow to run.
So instead of:
```
if(condition_a,
    do_a;
    return()
);
if(condition_b,
    do_b;
    return()
);
```
and etc., you can do:
```
if(condition_a,
    do_a,
    condition_b,
    do_b,
    last_thing
);
```

Reply:
Optimization is complete
thanks for your advice

os: Sorry, my English is not that good